includes/class-wc-stripe-database-cache-prefetch.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * Class WC_Stripe_Database_Cache_Prefetch
+ *
+ * This class is responsible for prefetching cache keys.
+ */
+class WC_Stripe_Database_Cache_Prefetch {
+	/**
+	 * The action used for the asynchronous cache prefetch code.
+	 *
+	 * @var string
+	 */
+	public const ASYNC_PREFETCH_ACTION = 'wc_stripe_database_cache_prefetch_async';
+
+	/**
+	 * Configuration for cache prefetching.
+	 *
+	 * @var int[]
+	 */
+	protected const PREFETCH_CONFIG = [
+		WC_Stripe_Payment_Method_Configurations::CONFIGURATION_CACHE_KEY => 10,
+	];
+
+	/**
+	 * The prefix used for prefetch tracking options.
+	 *
+	 * @var string
+	 */
+	private const PREFETCH_OPTION_PREFIX = 'wcstripe_prefetch_';
+
+	/**
+	 * The singleton instance.
+	 */
+	private static ?WC_Stripe_Database_Cache_Prefetch $instance = null;
+
+	/**
+	 * Protected constructor to support singleton pattern.
+	 */
+	protected function __construct() {}
+
+	/**
+	 * Get the singleton instance.
+	 *
+	 * @return WC_Stripe_Database_Cache_Prefetch The singleton instance.
+	 */
+	public static function get_instance(): WC_Stripe_Database_Cache_Prefetch {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Check if the unprefixed cache key has prefetch enabled.
+	 *
+	 * @param string $key The unprefixed cache key to check.
+	 * @return bool True if the cache key can be prefetched, false otherwise.
+	 */
+	public function should_prefetch_cache_key( string $key ): bool {
+		return isset( self::PREFETCH_CONFIG[ $key ] ) && self::PREFETCH_CONFIG[ $key ] > 0;
```

Comment:
We could add a filter here to provide a way for merchants to disable the prefetch for a specific key if they want.